cvFit/src/app/macroCalculator/calculator.component.ts

```diff
@@ -12,110 +15,62 @@ interface weightGoal{
     templateUrl: './calculator.component.html',
     styleUrls: ['./calculator.component.css']
 })
-export class CalculatorComponent{
-    //variables for calculations
-    age: number | undefined;
-    gender: string | undefined; 
-    genders: string[] = ["Male", "Female"];
-    heightF: number | undefined;
-    heightI: number | undefined;
-    weight: number | undefined;
-    activity: number = 1; //part of dropdown
+export class CalculatorComponent implements OnInit{
+    constructor(private calcService: CalcService) { }
+    macroForm: FormGroup;
+    showResult: boolean = false;
     activityL: activityLevel[] = [{value: 1, viewValue: "No exercise"}, {value: 2, viewValue: "Exercise 1-3 times a week"}, {value: 3, viewValue: "Exercise 4-5 times a week"}, {value: 4, viewValue: "Exercise 6-7 times a week"}]
-    goal: number = 1; //part of dropdown
     goalW: weightGoal[] = [{value: 1, viewValue: "Weight loss"}, {value: 2, viewValue: "Maintain weight"}, {value: 3, viewValue: "Gain weight"}]
-
+    goal: number | undefined;
     //variables for results 
-    protein: number =0;
-    carbs: number =0;
-    fat: number =0;
-    dailyCals: number | undefined;
-    
-    showResult: boolean = false;
     bmr: number | undefined;
+    dailyCals: number | undefined;
+    goalResults: number[] | undefined;
+    carbs: number = 0;
+    fat: number = 0;
+    protein: number = 0;
+
+    ngOnInit(): void {
+        let age = new FormControl();
+        let gender = new FormControl();
+        let heightF = new FormControl();
+        let heightI = new FormControl();
+        let weight = new FormControl();
+        let activityLevel = new FormControl();
+        let weightGoal = new FormControl();
+        this.macroForm = new FormGroup({
+            age: age,
+            gender: gender,
+            heightF: heightF,
+            heightI: heightI,
+            weight: weight,
+            activityLevel: activityLevel,
+            weightGoal: weightGoal
+        })
+    }
     
-    submit(formValues: { age: number; gender: string; heightF: number; heightI: number; weight: number; activity: number; goal: number; }){
-        
+    submit(){;
+        this.goal = this.macroForm.get('weightGoal').value //used in html (not sure how to access the form value in the html)
```

Comment:
You can access it like this:
`macroForm.controls.weightGoal.value` or `macroForm.controls[variableWithControlName].value`